examples/multicall_busybox.rs

```diff
@@ -5,27 +5,39 @@ use clap::{App, AppSettings, Arg};
 fn main() {
     let app = App::new(env!("CARGO_CRATE_NAME"))
         .setting(AppSettings::ArgRequiredElseHelp)
-        .arg(
-            Arg::new("install")
-                .long("install")
-                .help("Install hardlinks for all subcommands in path")
-                .exclusive(true)
-                .takes_value(true)
-                .default_missing_value("/usr/local/bin")
-                .use_delimiter(false),
+        .subcommand(
+            App::new("busybox")
+                .arg(
+                    Arg::new("install")
+                        .long("install")
+                        .help("Install hardlinks for all subcommands in path")
+                        .exclusive(true)
+                        .takes_value(true)
+                        .default_missing_value("/usr/local/bin")
+                        .use_delimiter(false),
+                )
+                .subcommand(App::new("true").about("does nothing successfully"))
+                .subcommand(App::new("false").about("does nothing unsuccessfully")),
         )
         .subcommand(App::new("true").about("does nothing successfully"))
         .subcommand(App::new("false").about("does nothing unsuccessfully"));
 
     let app = app.setting(AppSettings::Multicall);
     let matches = app.get_matches();
-    if matches.occurrences_of("install") > 0 {
-        unimplemented!("Make hardlinks to the executable here");
-    }
 
-    match matches.subcommand_name() {
-        Some("true") => exit(0),
-        Some("false") => exit(1),
-        _ => unreachable!("parser should ensure only valid subcommand names are used"),
-    }
+    exit(match matches.subcommand() {
+        Some(("busybox", cmd)) => {
+            if cmd.occurrences_of("install") > 0 {
+                unimplemented!("Make hardlinks to the executable here");
+            }
+            match cmd.subcommand() {
+                Some(("false", _)) => 1,
+                Some(("true", _)) => 0,
+                _ => 127,
+            }
+        }
+        Some(("false", _)) => 1,
+        Some(("true", _)) => 0,
```

Comment:
Train of thought: It'd be nice to de-duplicate the actual functionality.  matches don't quite work well for that.  We could just have the function recurse on itself and rely on the correctness of App to make sure we correctly recurse.  The main alternative I can think of is if we don't use match but an if-else ladder (which is what the macro generates)

Thoughts?

Reply:
Match works fairly well for `clap_derive` so it didn't occur to me to see how well an if-else ladder would fit.
I'll play with it a bit to see what works best, after I've addressed the other issues.

Reply:
I like if-else ladder for hand-coded parsers.